src/Common/CARESPTRResolver.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "DNSPTRResolver.h"
+
+using ares_channel = struct ares_channeldata *;
+
+namespace DB {
+    class CARESPTRResolver : public DNSPTRResolver {
+    public:
+        CARESPTRResolver();
+        ~CARESPTRResolver() override;
+
+        std::vector<std::string> resolve(const std::string & ip) override;
+
+    private:
+        void init();
+        void deinit();
+        void wait();
+
+        void resolve(const std::string & ip, std::vector<std::string> & response);
+
+        std::unique_ptr<ares_channel> channel;
```

Comment:
So it's unique pointer to raw pointer? Why do we need it?

Reply:
Pointer vs non-pointer:
We need it to be a pointer in order to use forward declaration and avoid including the header. That helps with the `reserved-identifier` and other build problems.

Reply:
That works indeed